app/models/manageiq/providers/azure/refresh_helper_methods.rb

```diff
@@ -18,10 +18,22 @@ def resource_uid(*keys)
     keys.join('\\')
   end
 
-  def gather_data_for_this_region(arm_service, method = "list")
-    resource_groups.collect do |resource_group|
-      arm_service.send(method, resource_group.name)
-    end.flatten
+  # For those resources without a location, default to the location of
+  # their resource group.
+  #
+  def gather_data_for_this_region(arm_service, method_name = 'list_all')
+    if method_name.to_s == 'list_all'
+      arm_service.send(method_name).select do |resource|
+        resource.location == @ems.provider_region
```

Comment:
The assumption is that if you use list_all, it has a location. Otherwise, I would have to do extra hoop jumping that I felt was unnecessary.